### DISCUSSION TITLE
Mutate data before creating

### DISCUSSION THREAD

**Discussion by jesus9314** [GuavaCZ/calendar#26] [Q&A] [answered]
How to mutate the data before send a form with getSchema method, same way to mutateFormDataBeforeCreate() from filament?

[Customizing data before saving](https://filamentphp.com/docs/3.x/panels/resources/creating-records#customizing-data-before-saving)

**Comment by lukas-frey** [comment, marked as answer]
You need to override the action, where you want to mutate the data for. So for example, if you want to mutate the form data in the `edit` action, you need to override the `editAction()` method from the calendar widget class:

```php
public function editAction(): Action
    {
        return parent::editAction()
            ->mutateFormDataUsing(function ($data) {
                // mutate data here
            })
        ;
    }
```
